packages/cli/commands/add/utils.ts

```diff
@@ -57,27 +57,45 @@ export function readFile(cwd: string, filePath: string): string {
 	return text;
 }
 
+const checkInstallNeeded = (atLeastVersion: string, version: string | undefined) => {
+	if (!version) return true;
+	const atLeastVersionStripped = atLeastVersion.replace('^', '');
+	const versionStripped = version.replace('^', '');
+	return !isVersionAtLeast(versionStripped, atLeastVersionStripped);
+};
```

Comment:
i think we may be doing too much with the inclusion of version checking. dependency versions can get very complicated with tons of edge-cases, so i'm not sure if this is really worth the effort. (note: we can get away with it with `@types/node` since we're targeting a very specific dependency version that we'll always know the shape/type of. we don't really have that guarantee with other 3rd party deps)

perhaps we should just show the install prompt only if there are any dependencies to be installed? although, im not sure how useful that'd be since almost every add-on tends to add some sort of new dependency

Reply:
> very complicated with tons of edge-cases

Don't tell me 😅, spent hours there before.

---

The idea is that `checkInstallNeeded` _(and `isVersionAtLeast` under)_, do it's best for simple case, and if not will fallback with prompting. So I would say that it's helping "normal" cases and fallback to todays way.

I would totally see "Community Add-On" for scaffolding code (with a dep).

---

But you are already the second one saying that it might not worth the effort. _(It's good to slow down my ideas from time to time :p)_
I still think it's a good addition as the fallback is like today. But yes, it's a bit of code, maintenance, surface, ...